internal/checker/utilities.go

```diff
@@ -1218,8 +1218,84 @@ func isValidNumberString(s string, roundTripOnly bool) bool {
 	return !n.IsNaN() && !n.IsInf() && (!roundTripOnly || n.String() == s)
 }
 
-func isValidBigIntString(s string, roundTripOnly bool) bool {
-	return false // !!!
+func isValidBigIntString(s string, roundtrip bool) bool {
+	if s == "" {
+		return false
+	}
+
+	// Create a scanner and set the text to the input string with an 'n' suffix
+	scan := scanner.NewScanner("", scanner.ScriptTargetESNext, false /*skipTrivia*/)
+	text := s + "n"
+	scan.SetText(text)
+	
+	// Scan the first token
+	token := scan.Scan()
```

Comment:
Nit, but it'd be helpful if this were still named `result` to match the origianl code.

Reply:
no problem, good catch... updated